app/policies/api/main_api_policy.rb

```diff
@@ -4,6 +4,7 @@ class MainApiPolicy < ApplicationPolicy
     default_rule :manage?
 
     skip_pre_check :role_exists?
+    skip_pre_check :view_hidden_course?
```

Comment:
I know you mentioned skipping the new policy wherever we skip the `role_exists?` but in this case, this might lead to students being able to access a hidden course through the api. I would like your opinion if that is alright or if the precheck should not be skipped in this instance or if it should be skipped elsewhere.

Reply:
added a safe navigation operator to the `:view_hidden_course?` pre-check so that it will only deny access if the role exists and is a student and the course is hidden